bundles/specmate-migration/src/com/specmate/migration/internal/services/MigratorService.java

```diff
@@ -84,33 +87,44 @@ private void closeConnection() throws SpecmateException {
 
 	@Override
 	public boolean needsMigration() throws SpecmateException {

```

Comment:
in case of an exception or if you are unable to initiate the db connection I wouldn't return false, but instead throw an exception to signal that something unexpected happened. Handling the exception is then the job of the client.